lib/ecto/type.ex

```diff
@@ -152,7 +150,7 @@ defmodule Ecto.Type do
   """
   @callback equal?(term, term) :: boolean
 
-  @optional_callbacks [equal?: 2]
+  @optional_callbacks [cast: 1, dump: 1, load: 1, equal?: 2]
```

Comment:
I've spoken with @josevalim, we're reverting this change because the default behaviour of passing through the value is too relaxed. E.g. if people wrap a string they should use `is_binary` guards for all calback implementations.

@michalmuskala we talked a while ago about limiting type boilerplate, do you have some ideas?